### ISSUE TITLE
Error: ConnectFailure (SocketException: Network is unreachable)

### ISSUE THREAD

**Issue by rotorgames** [couchbase/couchbase-lite-net#664]
Hi friends. I use Couchbase lite for Xamarin.Android.

I created two replication (push and pull).

```csharp
var username = "username";

_pull = Db.CreatePullReplication(new Uri(Config.ServerUrl));
_push = Db.CreatePushReplication(new Uri(Config.ServerUrl));

foreach (var replication in new[] { _pull, _push })
{
    replication.Channels = new[] { username };
    var date = DateTime.Parse("2017-06-09T00:20:46.163683541-04:00");
    replication.SetCookie("SyncGatewaySession", "68d3006a3cbd1d8e4d3acf7faa66e7f94974cd64", "/", date, false, false);
    replication.Continuous = true;
}

_pull.Start();
_push.Start();
```

If internet connection on device, disabled. Couchbase throw error:

```
Xamarin caused by: md52ce486a14f4bcd95899665e9d932190b.JavaProxyThrowable: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.AggregateException: One or more errors occurred. ---> System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
  at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000bc] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net.Sockets/Socket.cs:1235 
  at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0019b] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/WebConnection.cs:213 
  --- End of inner exception stack trace ---
  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x0005e] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/HttpWebRequest.cs:944 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (IAsyncResult iar, System.Func`2 endFunction, System.Action`1 endAction, System.Threading.Tasks.Task`1 promise, Boolean requiresSynchronization) [0x00014] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/FutureFactory.cs:552 
  --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (Boolean includeTaskCanceledExceptions) [0x00014] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Task.cs:2168 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (Boolean waitCompletionNotification) [0x00034] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Future.cs:568 
  at System.Threading.Tasks.Task`1[TResult].get_Result () [0x00000] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Future.cs:535 
  at Couchbase.Lite.Replicator.Puller.<StartChangeTracker>m__2 (System.Threading.Tasks.Task`1 t) [0x00000] in /Users/borrrden/Development/couchbase-lite-net/src/Couchbase.Lite.Shared/Replication/Puller.cs:174 
  at System.Threading.Tasks.ContinuationTaskFromResultTask`1[TAntecedentResult].InnerInvoke () [0x00027] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/TaskContinuation.cs:159 
  at System.Threading.Tasks.Task.Execute () [0x00016] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Task.cs:2523 
---> (Inner Exception #0) System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
  at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000bc] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net.Sockets/Socket.cs:1235 
  at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0019b] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/WebConnection.cs:213 
  --- End of inner exception stack trace ---
  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x0005e] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/HttpWebRequest.cs:944 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (IAsyncResult iar, System.Func`2 endFunction, System.Action`1 endAction, System.Threading.Tasks.Task`1 promise, Boolean requiresSynchronization) [0x00014] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/FutureFactory.cs:552 <---

  --- End of inner exception stack trace ---
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
  at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000bc] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net.Sockets/Socket.cs:1235 
  at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0019b] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/WebConnection.cs:213 
  --- End of inner exception stack trace ---
  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x0005e] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/HttpWebRequest.cs:944 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (IAsyncResult iar, System.Func`2 endFunction, System.Action`1 endAction, System.Threading.Tasks.Task`1 promise, Boolean requiresSynchronization) [0x00014] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/FutureFactory.cs:552 
  --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (Boolean includeTaskCanceledExceptions) [0x00014] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Task.cs:2168 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (Boolean waitCompletionNotification) [0x00034] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Future.cs:568 
  at System.Threading.Tasks.Task`1[TResult].get_Result () [0x00000] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Future.cs:535 
  at Couchbase.Lite.Replicator.Puller.<StartChangeTracker>m__2 (System.Threading.Tasks.Task`1 t) [0x00000] in /Users/borrrden/Development/couchbase-lite-net/src/Couchbase.Lite.Shared/Replication/Puller.cs:174 
  at System.Threading.Tasks.ContinuationTaskFromResultTask`1[TAntecedentResult].InnerInvoke () [0x00027] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/TaskContinuation.cs:159 
  at System.Threading.Tasks.Task.Execute () [0x00016] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/Task.cs:2523 
---> (Inner Exception #0) System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
  at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000bc] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net.Sockets/Socket.cs:1235 
  at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0019b] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/WebConnection.cs:213 
  --- End of inner exception stack trace ---
  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x0005e] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/mcs/class/System/System.Net/HttpWebRequest.cs:944 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (IAsyncResult iar, System.Func`2 endFunction, System.Action`1 endAction, System.Threading.Tasks.Task`1 promise, Boolean requiresSynchronization) [0x00014] in /Users/builder/data/lanes/3236/ee215fc9/source/mono/external/referencesource/mscorlib/system/threading/Tasks/FutureFactory.cs:552 <---
<---
```

Please, help me. How catch replication error?

Couchbase Lite 1.2.1.1
Couchbase.Lite.Storage.SystemSQLite 1.2.1.1
Android 5.1 (device and emulator)

**Comment by borrrden** [comment]
In that case, the Changed event should fire indicating that the replicator has changed states.  It will also contain any relevant error info.

**Comment by rotorgames** [comment]
@borrrden Yes, Changed event in replication invocing. I tried to sign up for Changes event, but still a mistake killing my application

**Comment by borrrden** [comment]
Oh, I misunderstood.  This error is crashing the application?  I thought it was just being logged.  It looks like the application will suicide if a task gets finalized without checking its exception property....there is one thing that you can do at the app level though:

https://msdn.microsoft.com/en-us/library/system.threading.tasks.taskscheduler.unobservedtaskexception(v=vs.110).aspx

Override this method, and call `SetObserved(true)` on the arguments you get.  I will also double check that our internal task schedulers check the exception for each task they process.

**Comment by rotorgames** [comment]
@borrrden Hi friend, `TaskScheduler.UnobservedTaskException` never call, also `AndroidEnvironment.UnhandledExceptionRaiser` never call. I'm can not catch error.

**Comment by rotorgames** [comment]
@borrrden Also new crash

```
caused.AggregateException(s)
Couchbase.Lite.Internal.WebSocketChangeTracker.<OnClose>m__0(Task t)WebSocketChangeTracker.cs:110
System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()TaskContinuation.cs:63
System.Threading.Tasks.Task.Execute()Task.cs:2502
--- End of inner exception stack trace ---
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
Couchbase.Lite.Internal.WebSocketChangeTracker.<OnClose>m__0(Task t)WebSocketChangeTracker.cs:110
System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()TaskContinuation.cs:63
System.Threading.Tasks.Task.Execute()Task.cs:2502
```

**Comment by borrrden** [comment]
Are you saying that the new exception occurs instead of the old one?  The place that the exception occurred for the other stack trace has been removed.

**Comment by rotorgames** [comment]
@borrrden No. The old error still exists. If you run the application from internet disconnections, throw this exception:

```
WebSocketChangeTracker.<OnClose>m__0(System.Threading.Tasks.Task t)
android.runtime.JavaProxyThrowable: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.NullReferenceException: Object reference not set to an instance of an object.
WebSocketChangeTracker.cs, line 110

caused.AggregateException(s)
Couchbase.Lite.Internal.WebSocketChangeTracker.<OnClose>m__0(Task t)WebSocketChangeTracker.cs:110
System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()TaskContinuation.cs:63
System.Threading.Tasks.Task.Execute()Task.cs:2502
--- End of inner exception stack trace ---
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
Couchbase.Lite.Internal.WebSocketChangeTracker.<OnClose>m__0(Task t)WebSocketChangeTracker.cs:110
System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()TaskContinuation.cs:63
System.Threading.Tasks.Task.Execute()Task.cs:2502
```

**Comment by borrrden** [comment]
But it's only with the new stack trace right?  Not the old one right?  I put in a guard against this so give it a try with the latest build.

**Comment by rotorgames** [comment]
@borrrden Thanks for the quick response. The latest build 118?

**Comment by borrrden** [comment]
It's now 122.

**Comment by rotorgames** [comment]
@borrrden Thank you. Application don't crashes now. But, hockeyapp pull message with error. Maybe, crash throw when application quit.

```
caused.AggregateException(s)
System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
System.Collections.Generic.List<T>.Enumerator.MoveNextRare()
System.Collections.Generic.List<T>.Enumerator.MoveNext()
System.Linq.Enumerable.All<TSource>(IEnumerable<_> source, Func<_,_> predicate)
Couchbase.Lite.Database.AddActiveReplication(Replication replication)
Couchbase.Lite.Replication.StartInternal()
Couchbase.Lite.Replication.<InitializeStateMachine>m__6(Transition transition)
Stateless.StateMachine<TState, TTrigger>.StateConfiguration.<>c__DisplayClass16_0.<OnEntry>b__0(Transition t, object[] args)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.ExecuteEntryActions(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.Enter(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.InternalFire(TTrigger trigger, object[] args)
Stateless.StateMachine<TState, TTrigger>.Fire(TTrigger trigger)
Couchbase.Lite.Replication.<FireTrigger>c__AnonStorey0.<>m__0()
--- End of inner exception stack trace ---
---> (Inner Exception #0) System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
System.Collections.Generic.List<T>.Enumerator.MoveNextRare()
System.Collections.Generic.List<T>.Enumerator.MoveNext()
System.Linq.Enumerable.All<TSource>(IEnumerable<_> source, Func<_,_> predicate)
Couchbase.Lite.Database.AddActiveReplication(Replication replication)
Couchbase.Lite.Replication.StartInternal()
Couchbase.Lite.Replication.<InitializeStateMachine>m__6(Transition transition)
Stateless.StateMachine<TState, TTrigger>.StateConfiguration.<>c__DisplayClass16_0.<OnEntry>b__0(Transition t, object[] args)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.ExecuteEntryActions(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.Enter(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.InternalFire(TTrigger trigger, object[] args)
Stateless.StateMachine<TState, TTrigger>.Fire(TTrigger trigger)
Couchbase.Lite.Replication.<FireTrigger>c__AnonStorey0.<>m__0()
```

**Comment by rotorgames** [comment]
@borrrden And new error when do get request in database:

```csharp
private View AllFromTypeView => Db.GetView("categories-AllFromType");

...

using (var query = AllFromTypeView.CreateQuery())
{
    query.StartKey = new object[] {SyncService.Username, type};
    query.EndKey = new object[] { SyncService.Username, type };

    var rows = await query.RunAsync();

    // Convert Document to C# class model
    return rows.ToList<CategoryTable>(); 
}

...

AllFromTypeView.SetMap(document, emit) =>
{
     string owner= // this username in document;
     enum type = // this enum in document;
     emit(new object[] { owner, type }, null);
}, "1");

```

```
caused.AggregateException(s)
System.Net.Http.HttpMessageHandler.Dispose()
System.Net.Http.DelegatingHandler.Dispose(bool disposing)
System.Net.Http.HttpMessageHandler.Dispose()
System.Net.Http.HttpMessageInvoker.Dispose(bool disposing)
System.Net.Http.HttpClient.Dispose(bool disposing)
System.Net.Http.HttpMessageInvoker.Dispose()
Couchbase.Lite.Internal.Leasable<T>.DisposeInternal()
Couchbase.Lite.Internal.Leasable<T>.Dispose()
Couchbase.Lite.CouchbaseLiteHttpClient.Dispose()
Couchbase.Lite.Internal.Leasable<T>.DisposeInternal()
Couchbase.Lite.Internal.Leasable<T>.Dispose()
Couchbase.Lite.Internal.RemoteSession.Dispose()
Couchbase.Lite.Replicator.Puller.StopGraceful()
Couchbase.Lite.Replication.<InitializeStateMachine>m__B(Transition transition)
Stateless.StateMachine<TState, TTrigger>.StateConfiguration.<>c__DisplayClass16_0.<OnEntry>b__0(Transition t, object[] args)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.ExecuteEntryActions(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.Enter(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.InternalFire(TTrigger trigger, object[] args)
Stateless.StateMachine<TState, TTrigger>.Fire(TTrigger trigger)
Couchbase.Lite.Replication.<FireTrigger>c__AnonStorey0.<>m__0()
--- End of inner exception stack trace ---
---> (Inner Exception #0) System.MethodAccessException: Method `System.Net.ServicePointManager:CloseConnectionGroup (string)' is inaccessible from method `System.Net.Http.HttpClientHandler:Dispose (bool)'
System.Net.Http.HttpMessageHandler.Dispose()
System.Net.Http.DelegatingHandler.Dispose(bool disposing)
System.Net.Http.HttpMessageHandler.Dispose()
System.Net.Http.HttpMessageInvoker.Dispose(bool disposing)
System.Net.Http.HttpClient.Dispose(bool disposing)
System.Net.Http.HttpMessageInvoker.Dispose()
Couchbase.Lite.Internal.Leasable<T>.DisposeInternal()
Couchbase.Lite.Internal.Leasable<T>.Dispose()
Couchbase.Lite.CouchbaseLiteHttpClient.Dispose()
Couchbase.Lite.Internal.Leasable<T>.DisposeInternal()
Couchbase.Lite.Internal.Leasable<T>.Dispose()
Couchbase.Lite.Internal.RemoteSession.Dispose()
Couchbase.Lite.Replicator.Puller.StopGraceful()
Couchbase.Lite.Replication.<InitializeStateMachine>m__B(Transition transition)
Stateless.StateMachine<TState, TTrigger>.StateConfiguration.<>c__DisplayClass16_0.<OnEntry>b__0(Transition t, object[] args)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.ExecuteEntryActions(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.Enter(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.InternalFire(TTrigger trigger, object[] args)
Stateless.StateMachine<TState, TTrigger>.Fire(TTrigger trigger)
Couchbase.Lite.Replication.<FireTrigger>c__AnonStorey0.<>m__0()
```

**Comment by borrrden** [comment]
You are still just starting a replication while the device is in airplane mode?  I have the same device and I cannot reproduce.  That last message is really weird because it's basically saying your version of mono is different than mine and so your methods are different.  What versions of mono and Xamarin.Android do you have installed?

**Comment by rotorgames** [comment]
@borrrden 

Xamarin.VS 4.1.1
Xamarin.Android 6.1

Previouse error throw only on Android 6.0. Maybe android cache data files, and restore it after reinstall app.
[Auto Backup](https://developer.android.com/training/backup/autosyncapi.html)

When I disabled Auto Backup, error missing. But this error remained on all devices.

```
caused.AggregateException(s)
System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
System.Collections.Generic.List<T>.Enumerator.MoveNextRare()
System.Collections.Generic.List<T>.Enumerator.MoveNext()
System.Linq.Enumerable.All<TSource>(IEnumerable<_> source, Func<_,_> predicate)
Couchbase.Lite.Database.AddActiveReplication(Replication replication)
Couchbase.Lite.Replication.StartInternal()
Couchbase.Lite.Replication.<InitializeStateMachine>m__6(Transition transition)
Stateless.StateMachine<TState, TTrigger>.StateConfiguration.<>c__DisplayClass16_0.<OnEntry>b__0(Transition t, object[] args)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.ExecuteEntryActions(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.Enter(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.InternalFire(TTrigger trigger, object[] args)
Stateless.StateMachine<TState, TTrigger>.Fire(TTrigger trigger)
Couchbase.Lite.Replication.<FireTrigger>c__AnonStorey0.<>m__0()
--- End of inner exception stack trace ---
---> (Inner Exception #0) System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
System.Collections.Generic.List<T>.Enumerator.MoveNextRare()
System.Collections.Generic.List<T>.Enumerator.MoveNext()
System.Linq.Enumerable.All<TSource>(IEnumerable<_> source, Func<_,_> predicate)
Couchbase.Lite.Database.AddActiveReplication(Replication replication)
Couchbase.Lite.Replication.StartInternal()
Couchbase.Lite.Replication.<InitializeStateMachine>m__6(Transition transition)
Stateless.StateMachine<TState, TTrigger>.StateConfiguration.<>c__DisplayClass16_0.<OnEntry>b__0(Transition t, object[] args)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.ExecuteEntryActions(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.StateRepresentation.Enter(Transition transition, object[] entryArgs)
Stateless.StateMachine<TState, TTrigger>.InternalFire(TTrigger trigger, object[] args)
Stateless.StateMachine<TState, TTrigger>.Fire(TTrigger trigger)
Couchbase.Lite.Replication.<FireTrigger>c__AnonStorey0.<>m__0()
```

**Comment by borrrden** [comment]
That error is something I can do something about.  I will put in a commit to guard access to that collection soon.

**Comment by rotorgames** [comment]
@borrrden Thank you.

**Comment by borrrden** [comment]
Feel free to try again with the latest build.  I guarded access to that collection now.

**Comment by rotorgames** [comment]
@ latest build 130?

**Comment by borrrden** [comment]
Yes, or any subsequent ones that come out between now and when you try.

**Comment by rotorgames** [comment]
@borrrden New error.

Android 6.0.1
Nexus 5

```

caused.AggregateException(s)
--- End of inner exception stack trace ---
System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions)
System.Threading.Tasks.Task<TResult>.GetResultCore(bool waitCompletionNotification)
System.Threading.Tasks.Task<TResult>.get_Result()
Couchbase.Lite.CouchbaseLiteHttpClient.<SendAsync>c__AnonStorey2.<>m__1(Task<_> t)
System.Threading.Tasks.ContinuationResultTaskFromResultTask<TAntecedentResult, TResult>.InnerInvoke()
System.Threading.Tasks.Task.Execute()
---> (Inner Exception #0) System.Threading.Tasks.TaskCanceledException: A task was canceled.<---
--- End of inner exception stack trace ---
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
--- End of inner exception stack trace ---
System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions)
System.Threading.Tasks.Task<TResult>.GetResultCore(bool waitCompletionNotification)
System.Threading.Tasks.Task<TResult>.get_Result()
Couchbase.Lite.CouchbaseLiteHttpClient.<SendAsync>c__AnonStorey2.<>m__1(Task<_> t)
System.Threading.Tasks.ContinuationResultTaskFromResultTask<TAntecedentResult, TResult>.InnerInvoke()
System.Threading.Tasks.Task.Execute()
---> (Inner Exception #0) System.Threading.Tasks.TaskCanceledException: A task was canceled.<---
<---

```

**Comment by borrrden** [comment]
@rotorgames does it crash or just show that error?  That error is something I would expect from an unreachable server.

**Comment by rotorgames** [comment]
@borrrden crash, only on android 6

**Comment by borrrden** [comment]
Do you have any other logs?  I have the exact same hardware and version as you and I've never once seen this.

**Comment by rotorgames** [comment]
@borrrden There is only this.

```
Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
android.runtime.JavaProxyThrowable: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.AggregateException: One or more errors occurred. ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
```

You are tested on LGE NEXUS 5, Android 6.0.1?

**Comment by borrrden** [comment]
LGE Nexus 5 Android 6.0.  The unit tests that use network do not crash when the phone is in airplane mode.

**Comment by rotorgames** [comment]
@borrrden I tested in the 3G / 4G / WiFi mode. Used Sync Gateway. App crashes.

With disabled internet application works fine.

**Comment by borrrden** [comment]
I'm going to need a reproduction case for this then so we can get on the same page.  Can you confirm that an app that does just what you posted in the beginning has the same issue?

**Comment by rotorgames** [comment]
@borrrden I do not quite understand the question.

The application is the same. All I did was updated Couchbase.Lite.

Appendix always crash at the inclusion of the Internet. Previously, it happened on all devices now only on LGE NEXUS 5, Android 6.0.1.

I create a new issue?

**Comment by borrrden** [comment]
What I mean is, can you isolate the issue down from "your application" to a simple project that can reproduce the issue for you?  That way I can run it over here and see if the incorrect behavior happens.

**Comment by rotorgames** [comment]
@borrrden OK. I will create a repository and let you know. Thank you

**Comment by borrrden** [comment]
@rotorgames What is the story on the repo?

**Comment by borrrden** [comment]
Closing this ticket for lack of activity, but this might be helped by some of the `issue/718` branch work

**Comment by marcusturewicz** [comment]
I am getting this error. Happens when internet is lost after replication works correctly.

```

    01-06 16:34:31.440 E/mono    (23265): Unhandled Exception:
    01-06 16:34:31.440 E/mono    (23265): System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000cb] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System/System.Net.Sockets/Socket.cs:1313 
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0019b] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System/System.Net/WebConnection.cs:195 
    01-06 16:34:31.440 E/mono    (23265):    --- End of inner exception stack trace ---
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x0005e] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System/System.Net/HttpWebRequest.cs:1023 
    01-06 16:34:31.440 E/mono    (23265):   at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x00014] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/threading/Tasks/FutureFactory.cs:550 
    01-06 16:34:31.440 E/mono    (23265): --- End of stack trace from previous location where exception was thrown ---
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.Http.HttpClientHandler+<SendAsync>c__async0.MoveNext () [0x003ce] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System.Net.Http/System.Net.Http/HttpClientHandler.cs:379 
    01-06 16:34:31.440 E/mono    (23265): --- End of stack trace from previous location where exception was thrown ---
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.
```

**Comment by tura08** [comment]
I am getting this error. Happens when internet is lost after replication works correctly. Using Xamarin Android, latest CBL 1.4 build 14

    01-06 16:34:31.440 E/mono    (23265): Unhandled Exception:
    01-06 16:34:31.440 E/mono    (23265): System.Net.WebException: Error: ConnectFailure (Network is unreachable) ---> System.Net.Sockets.SocketException: Network is unreachable
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) [0x000cb] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System/System.Net.Sockets/Socket.cs:1313 
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.WebConnection.Connect (System.Net.HttpWebRequest request) [0x0019b] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System/System.Net/WebConnection.cs:195 
    01-06 16:34:31.440 E/mono    (23265):    --- End of inner exception stack trace ---
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x0005e] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System/System.Net/HttpWebRequest.cs:1023 
    01-06 16:34:31.440 E/mono    (23265):   at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x00014] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/threading/Tasks/FutureFactory.cs:550 
    01-06 16:34:31.440 E/mono    (23265): --- End of stack trace from previous location where exception was thrown ---
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
    01-06 16:34:31.440 E/mono    (23265):   at System.Net.Http.HttpClientHandler+<SendAsync>c__async0.MoveNext () [0x003ce] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/System.Net.Http/System.Net.Http/HttpClientHandler.cs:379 
    01-06 16:34:31.440 E/mono    (23265): --- End of stack trace from previous location where exception was thrown ---
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
    01-06 16:34:31.440 E/mono    (23265):   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/data/lanes/3511/501e63ce/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.
